File: app/src/main/java/com/vmenon/mpo/core/persistence/MPODatabase.java
package com.vmenon.mpo.core.persistence;

import androidx.room.*;

import com.vmenon.mpo.api.Episode;
import com.vmenon.mpo.api.Show;

@Database(entities = {Show.class, Episode.class}, version = 1)
@TypeConverters({Converters.class})
public abstract class MPODatabase extends RoomDatabase {
    public abstract ShowDao showDao();
    public abstract EpisodeDao episodeDao();
}

File: app/src/main/java/com/vmenon/mpo/core/persistence/MPODatabase.kt
package com.vmenon.mpo.core.persistence

import androidx.room.*
import com.vmenon.mpo.model.*

@Database(
    entities = [
        ShowModel::class,
        EpisodeModel::class,
        ShowSearchModel::class,
        ShowSearchResultsModel::class,
        DownloadModel::class
    ],
    version = 1
)
@TypeConverters(Converters::class)
abstract class MPODatabase : RoomDatabase() {
    abstract fun showDao(): ShowDao
    abstract fun episodeDao(): EpisodeDao
    abstract fun showSearchResultsDao(): ShowSearchResultDao
    abstract fun downloadDao(): DownloadDao
}
